cognite/neat/_utils/text.py

```diff
@@ -12,22 +12,27 @@ def to_camel(string: str) -> str:
         camelCase of the input string.
 
     Examples:
-        >>> to_camel("a_b")
+        >>> to_camel_case("a_b")
         'aB'
-        >>> to_camel("ScenarioInstance_priceForecast")
+        >>> to_camel_case("ScenarioInstance_priceForecast")
         'scenarioInstancePriceForecast'
     """
-    string = re.sub(r"[\s_-]", "_", string)
+    return _to_camel_case(string, string == string.upper())
+
+
+def _to_camel_case(string, is_all_upper: bool):
+    string = re.sub(r"[^a-zA-Z0-9_]", "_", string)
```

Comment:
Making this much more strict. We only allow letters and numbers (underscore is used for appropriate splitting)